src/Client/HTTPClient.cpp

```diff
@@ -0,0 +1,97 @@
+#include <sstream>
+#include <QJsonObject>
+
+#include <Poco/URI.h>
+#include <Poco/Net/HTTPResponse.h>
+#include <Poco/StreamCopier.h>
+#include <Poco/Message.h>
+
+#include "Exception.h"
+#include "HTTPClient.h"
+
+void HTTPClient::set_token(const std::string& str_token)
+{
+    m_token = str_token;
+}
+
+void HTTPClient::set_auth_scheme(const std::string& str_auth_scheme)
+{
+    m_auth_scheme = str_auth_scheme;
+}
+
+void HTTPClient::SendRequest(
+    Poco::Net::HTTPRequest &request,
+    Poco::Net::HTTPClientSession &session,
+    const Net::Request &net_request)
+{
+    AddCredentials(request);
+    request.setContentType(net_request.content_type);
+    if (net_request.content_type == Net::CONTENT_TYPE_APPLICATION_JSON)
+    {
+        QByteArray byte_array = net_request.json_playload.toJson();
+        request.setContentLength(byte_array.size());
+        std::ostream& out_stream = session.sendRequest(request);
+        out_stream << net_request.json_playload.toJson().constData();
+    }
+    else if (net_request.content_type == Net::CONTENT_TYPE_PLAIN_TEXT)
+    {
+		session.sendRequest(request);
+    }
+    else if (net_request.content_type == Net::CONTENT_TYPE_EMPTY)
+    {
+		session.sendRequest(request);
+    }
+    else
+    {
+        throw Exception("Unsupported content type");
+    }
+}
+
+Net::Response HTTPClient::FormResponse(
+    const Poco::Net::HTTPResponse& response,
+    std::istream &received_stream)
```

Comment:
as per code convention, prefer & adjacent to type
```suggestion
    std::istream& received_stream)
```